test/libsolidity/semanticTests/operators/userDefined/operator_parameter_and_return_cleanup_between_calls.sol

```diff
@@ -0,0 +1,22 @@
+type U8 is uint8;
+using {yoloAdd as +, yoloDiv as /} for U8 global;
+
+function yoloAdd(U8 x, U8 y) pure returns (U8 z) {
+    assembly {
+        z := add(x, y) // Wrong! No cleanup.
```

Comment:
Just curious since I don't know - what happens in this case, i.e. what does `z` become? I would guess that `0xff` and `3` should overflow and make `z` equal to `2`, which should then work fine with `yoloDiv`, i.e. it should return `2` as the answer (`4/2`), but it doesn't. What kind of cleanup is performed with overflow?

Reply:
It's assembly so `z`, `x` and `y` are full 256-bit stack slots. There's no overflow for `add(0xff, 3)` because the result fits just fine into such a slot. You get `0x102` in that slot. You can access that whole value only via inline assembly though. If you try to access the variable from Solidity code, you'll see only `0x02` because the compiler will insert extra shifts and masking to hide the rest. This is only performed for a specific operation though. The original `z` on the stack is unaffected and still has these dirty bits.

The problem here is that you might expect values to be cleaned when they're passed into parameters or returned from functions. They're not. If you return `z` from `yoloAdd()` and then pass it into `yoloDiv()`, you'll still be able to see those dirty upper bits in inline assembly.